alica_engine/include/engine/PlanBase.h

```diff
@@ -14,91 +14,93 @@
 #include <typeinfo>
 #include "engine/IAlicaClock.h"
 #include "engine/RunningPlan.h"
+#include "engine/RuleBook.h"
 
-using namespace std;
 namespace alica
 {
-	class Plan;
-	class RuleBook;
-	class AlicaEngine;
-	class TeamObserver;
-	class IRoleAssignment;
-	class Logger;
-	class AuthorityManager;
-	class SyncModule;
-	class IAlicaCommunication;
-	class Task;
-	class State;
-	class EntryPoint;
-	class IAlicaClock;
-	class Assignment;
-	class StateCollection;
-	class AlicaEngine;
+    class Plan;
+    
+    class AlicaEngine;
+    class TeamObserver;
+    class IRoleAssignment;
+    class Logger;
+    class AuthorityManager;
+    class SyncModule;
+    class IAlicaCommunication;
+    class Task;
+    class State;
+    class EntryPoint;
+    class IAlicaClock;
+    class Assignment;
+    class StateCollection;
+    class AlicaEngine;
 
-	/**
-	 * A PlanBase holds the internal representation of the plan graph and issues all operations on it.
-	 * It is the most central object within the ALICA Engine.
-	 */
-	class PlanBase
-	{
-	public:
-		PlanBase(AlicaEngine* ae, Plan* masterplan);
-		~PlanBase();
-		condition_variable* getStepModeCV();
-		const shared_ptr<RunningPlan> getRootNode() const;
-		void setRootNode(shared_ptr<RunningPlan> rootNode);
-		void setRuleBook(RuleBook* ruleBook);
-		const ulong getloopInterval() const;
-		void setLoopInterval(ulong loopInterval);
-		void stop();
-		void start();
-		void addFastPathEvent(shared_ptr<RunningPlan> p);
-		shared_ptr<RunningPlan> getDeepestNode();
-		shared_ptr<RunningPlan> getRootNode();
-		Plan* getMasterPlan();
+    /**
+     * A PlanBase holds the internal representation of the plan graph and issues all operations on it.
+     * It is the most central object within the ALICA Engine.
+     */
+    class PlanBase
+    {
+    public:
+        PlanBase(AlicaEngine* ae, Plan* masterplan);
+        ~PlanBase();
+        std::condition_variable* getStepModeCV();
+        const std::shared_ptr<RunningPlan> getRootNode() const;
+        void setRootNode(std::shared_ptr<RunningPlan> rootNode);
+        const AlicaTime getloopInterval() const;
+        void setLoopInterval(ulong loopInterval);
+        void stop();
+        void start();
+        void addFastPathEvent(shared_ptr<RunningPlan> p);
+        std::shared_ptr<RunningPlan> getDeepestNode();
+        std::shared_ptr<RunningPlan> getRootNode();
 
-	private:
-		/**
-		 * List of RunningPlans scheduled for out-of-loop evaluation.
-		 */
-		queue<shared_ptr<RunningPlan>> fpEvents;
-		condition_variable* stepModeCV;
-//		void checkPlanBase(shared_ptr<RunningPlan> r);
+        /*const*/ Plan* getMasterPlan() const {return _masterPlan;}
```

Comment:
What is the purpose of the commented out const here?

Reply:
It should be const, as all model element should be at runtime. But that would cause a rather big set of change that i don't want to do in this PR.

Reply:
I am tempted to keep the implementation and interface separated here, despite this function's simplicity.

Reply:
But why? This allows the call site to inline the call.
It should be done to many more implementations.